mountpoint-s3-crt/src/auth/credentials.rs

```diff
@@ -42,6 +57,28 @@ impl CredentialsProvider {
 
         Ok(Self { inner })
     }
+
+    /// Creates the profile credential provider.
+    pub fn new_profile(allocator: &Allocator, options: CredentialsProviderProfileOptions) -> Result<Self, Error> {
+        auth_library_init(allocator);
+
+        let profile_name = options
+            .profile_name_override
+            .ok_or(Error::from(AWS_AUTH_CREDENTIALS_PROVIDER_PROFILE_SOURCE_FAILURE as i32))?;
```

Comment:
if we error on missing profile name, why make the field an Option?

Reply:
The is address the comment from james, which I don't think passed in profile_name reference should be `Option<&'a &str>` because in `s3_crt_client.rs` we have take the string out of `Option`

Reply:
I was wrong about this, actually -- the CRT requires `profile_name_override`, so we shouldn't make it an `Option`. Sorry!